deps/Makefile

```diff
@@ -32,6 +32,11 @@ include $(SRCDIR)/llvm-ver.make
 
 ## Some shared configuration options ##
 
+# don't pass makefile flags through by default
+# (which would often result in unintentionally overriding the flags we are trying to set here)
+MAKEOVERRIDES =
```

Comment:
most distros don't want to build the contents of deps.

this is more predictable than the current situation (which can lead to nasty surprises), although i entirely expect that more work will need to happen over time to address other cases as they come up (as mentioned in the commit)